### PR TITLE
Fix/common add staking repository

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#8693mek1k
#8693n8hd8
#8693kxrzg
#8693kxtgn
#8693kxtek

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-8693kxtgn Android. Don't allow to add staking proxy for current account](https://app.clickup.com/t/8693kxtgn)

**Comment by stepanLav** [comment]
Task linked: [CU-8693n8hd8 Android. Show add proxy only for proxy networks](https://app.clickup.com/t/8693n8hd8)

**Comment by stepanLav** [comment]
Task linked: [CU-8693kxtek Android. Make an updater for selected account to sync proxies](https://app.clickup.com/t/8693kxtek)

**Comment by stepanLav** [comment]
Task linked: [CU-8693kxrzg Android. Show difference between new and old deposit for add proxy](https://app.clickup.com/t/8693kxrzg)

File: feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validations/delegation/proxy/add/Declarations.kt [hunk 1]
```diff
@@ -67,22 +70,18 @@ fun AddStakingProxyValidationSystemBuilder.maximumProxies(
...
-) = enoughBalanceToPayProxyDeposit(
-    chain = { it.chain },
-    accountId = { it.proxiedAccountId },
-    newDeposit = { it.depositWithQuantity.deposit },
-    availableBalance = { it.asset.freeInPlanks - it.asset.frozenInPlanks },
-    error = { payload, maxUsable ->
+fun AddStakingProxyValidationSystemBuilder.enoughBalanceToPayDepositAndFee() = sufficientBalance(
+    fee = { it.fee },
+    amount = { it.asset.token.configuration.amountFromPlanks(it.deltaDeposit) },
+    available = { it.asset.free - it.asset.frozen },
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validations/delegation/proxy/add/Declarations.kt:76]: I suggest to use methods from `Asset.Companion` to calculate transferable. In particular we can use `TransferableMode.REGULAR.calculateTransferable(free, frozen, reserved)`. That way we will keep transferable & ed calculation logic in a single place

**Review by valentunn — APPROVED** [review]

File: feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validations/delegation/proxy/add/Declarations.kt [hunk 2]
```diff
@@ -57,7 +60,7 @@ fun AddStakingProxyValidationSystemBuilder.maximumProxies(
 ) = maximumProxiesNotReached(
     chain = { it.chain },
     accountId = { it.proxiedAccountId },
-    newProxiedQuantity = { it.depositWithQuantity.quantity },
+    newProxiedQuantity = { it.currentQuantity + 1 },
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validations/delegation/proxy/add/Declarations.kt:63]: It seems to me that this is not necessary. In any case, we use a similar approach in other validations

**Pull request merged**